RabbitMq/Consumer.php

```diff
@@ -281,25 +299,30 @@ private function chooseWaitTimeout()
              * Respect the idle timeout if it's set and if it's less than
              * the remaining allowed execution.
              */
-            if (
-                $this->getIdleTimeout()
+            if ($this->getIdleTimeout()
                 && $this->getIdleTimeout() < $allowedExecutionSeconds
             ) {
-                return array(
-                    'timeoutType' => self::TIMEOUT_TYPE_IDLE,
-                    'seconds' => $this->getIdleTimeout(),
-                );
+                $waitTimeout = $this->getIdleTimeout();
+            } else {
+                $waitTimeout = $allowedExecutionSeconds;
             }
+        } else {
+            $waitTimeout = $this->getIdleTimeout();
+        }
 
-            return array(
-                'timeoutType' => self::TIMEOUT_TYPE_GRACEFUL_MAX_EXECUTION,
-                'seconds' => $allowedExecutionSeconds,
-            );
+        if ($this->getTimeoutWait()) {
```

Comment:
Please check  if (!is_null($this->getTimeoutWait())) or  if ($this->getTimeoutWait() !== null) because it returns an integer value. Check also: https://scrutinizer-ci.com/g/eMAGTechLabs/RabbitMqBundle/inspections/d889e465-34c7-4f57-bea6-6300aa404c64/issues/files/RabbitMq/Consumer.php?status=new&orderField=path&order=asc&honorSelectedPaths=0